src/EFCore.PG/Storage/Internal/NpgsqlRelationalConnection.cs

```diff
@@ -35,6 +38,8 @@ protected override DbConnection CreateDbConnection()
                 conn.ProvideClientCertificatesCallback = ProvideClientCertificatesCallback;
             if (RemoteCertificateValidationCallback != null)
                 conn.UserCertificateValidationCallback = RemoteCertificateValidationCallback;
+            if (RemoteCertificateValidationCallback != null)
+                conn.ProvidePasswordCallback = ProvidePasswordCallback;
```

Comment:
There is a bug, You're checking `RemoteCertificateValidationCallback `, but setting `ProvidePasswordCallback`.

Reply:
sorry for the copypasta, ive fixed it up and force pushed so this is still a single commit